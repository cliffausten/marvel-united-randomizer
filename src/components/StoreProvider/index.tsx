import React from 'react';
import { createRootStore } from 'stores/RootStore';

export const storeContext = React.createContext(null);

type Props = {
  children: React.ReactNode;
};

export const StoreProvider: React.FC<Props> = ({ children }: Props) => {
  const store = createRootStore();

  return (
    <storeContext.Provider value={store}>{children}</storeContext.Provider>
  );
};

export default StoreProvider;
