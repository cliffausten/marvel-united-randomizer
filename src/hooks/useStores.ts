import React from 'react';
import { storeContext } from 'components/StoreProvider';
import RootStore from 'stores/RootStore';

export const useStores = (): RootStore => {
  const store = React.useContext(storeContext);
  if (!store) {
    throw new Error('useStores must be used within a StoreProvider.');
  }
  return store;
};

export default useStores;
