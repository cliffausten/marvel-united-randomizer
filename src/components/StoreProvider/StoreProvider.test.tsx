import React from 'react';
import { mount } from 'enzyme';
import StoreProvider from '.';
import useStores from 'hooks/useStores';
jest.mock('stores/RootStore');

const MockComponent: React.FC = () => {
  useStores();

  return <div>TEST_CONTENT</div>;
};

describe('StoreProvider', () => {
  it('Render', () => {
    const storeProvider = mount(
      <StoreProvider>
        <MockComponent />
      </StoreProvider>
    );

    expect(storeProvider).toMatchSnapshot();
  });

  it('Render without StoreProvider', () => {
    expect(() => {
      mount(<MockComponent />);
    }).toThrow(Error);
  });
});
