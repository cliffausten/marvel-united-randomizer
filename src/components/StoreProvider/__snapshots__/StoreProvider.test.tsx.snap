// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreProvider Render 1`] = `
<StoreProvider>
  <MockComponent>
    <div>
      TEST_CONTENT
    </div>
  </MockComponent>
</StoreProvider>
`;
